src/serial.c

```diff
@@ -0,0 +1,64 @@
+#include "serial.h"
+
+static mutex_t _send_lock;
+
+/**
+ * @brief Try to acquire the serial transfer lock.
+ *
+ * @return Whether the lock was acquired or not.
+ */
+bool serial_acquire_transfer_lock(void)
+{
+	return (bool)mutex_trylock(&_send_lock);
+}
+
+/**
+ * @brief Send data through serial.
+ *
+ * DMA is configured to read from `data` a number `size` of bytes. It then
+ * writes all those bytes to USART1 (Bluetooth).
+ *
+ * An interruption is generated when the transfer is complete.
+ *
+ * @param[in] data Data to send.
+ * @param[in] size Size (number of bytes) to send.
+ */
```

Comment:
I would add the description of the steps needed to send by serial on the function header.

Reply:
DMA is not simple, and it does change with respect to the Bulebule implementation (the API is different, not our issue).

I think it is required to read/know the actual implementation on the datasheet. It would be hard to explain better the actual implementation in the header than what we currently have (a couple of single-purpose configuration calls in the implementation).

Reply:
Ok